src/main/java/org/nuxeo/wopi/rest/FilesResource.java

```diff
@@ -217,6 +219,43 @@ protected Object lock(String fileId, String lock, String oldLock) {
         }
     }
 
+    /**
+     * Implements the GetLock operation.
+     * <p>
+     * See <a href="https://wopi.readthedocs.io/projects/wopirest/en/latest/files/GetLock.html"></a>.
+     */
+    protected Object getLock(String fileId) {
+        NuxeoPrincipal principal = (NuxeoPrincipal) request.getUserPrincipal();
+        try (CloseableCoreSession session = createCoreSession(principal)) {
+            DocumentRef ref = new IdRef(fileId);
+            if (!session.exists(ref)) {
+                return Response.status(Response.Status.NOT_FOUND).build();
+            }
+
+            DocumentModel doc = session.getDocument(ref);
+            Blob blob = getMainBlob(doc);
+            if (blob == null) {
+                return Response.status(Response.Status.NOT_FOUND).build();
+            }
+
+            if (!doc.isLocked()) {
+                response.addHeader("X-WOPI-Lock", "");
+                return Response.ok().build();
+            }
+
+            KeyValueService service = Framework.getService(KeyValueService.class);
+            KeyValueStore store = service.getKeyValueStore("wopi-locks");
+            String currentLock = store.getString(doc.getId());
```

Comment:
Note that we don't contribute explicitly a KeyValueStore, we leverage the default one. 
As it, we only have one contribution to switch all K/V from Mem to Redis/MongoDB.
We're used to suffix all of our keys for this reason.